### PR TITLE
Rebrand samples to `ArcGIS Maps SDK for .NET`

### PR DESCRIPTION
**Pull request by ZackAllen**
# Description

This PR updates all of our sample names to reflect the rebranding to ArcGIS Maps SDK for .NET.
[This blogpost](https://esriurl.com/maps-sdks-announcement) explains the upcoming change.

## Type of change

- [x] Other enhancement

**Platforms tested on**:

- [x] WPF .NET 6
- [x] WPF Framework
- [x] WinUI
- [x] Maui Android
- [ ] Maui iOS
- [ ] Maui MacCatalyst
- [x] Maui WinUI
- [x] UWP

## Checklist

- [x] Runs and compiles on all active platforms
- [x] Legacy platforms still compile and run (if applicable)
- [x] Branch is up to date with the latest main/v.next
- [x] All merge conflicts have been resolved
- [x] Self-review of changes
- [x] There are no warnings related to changes
- [x] No unrelated changes have been made to any other code or project files
- [x] `sample_sync.py` runs without making changes
- [ ] Codemaid and XAML styler extensions have been run on every changed file
- [ ] Code is commented with correct formatting
- [ ] All variable and method names are good and make sense
- [ ] There is no leftover commented code
- [ ] Screenshots are correct size and display in description tab

### CODE REVIEW

**Review by nCastle1 — APPROVED** [review]

File: src/MAUI/Maui.Samples/Samples/Layers/RasterLayerRasterFunction/RasterLayerRasterFunction.xaml.cs [hunk 1]
```diff
@@ -112,7 +112,7 @@ private async Task Initialize()
                 // Zoom the map to the extent of the image service raster (which also the extent of the raster layer)
                 await MyMapView.SetViewpointGeometryAsync(myArcGISImageServiceInfo.FullExtent);
 
-                // NOTE: The sample zooms to the extent of the ImageServiceRaster. Currently the ArcGIS Runtime does not
+                // NOTE: The sample zooms to the extent of the ImageServiceRaster. Currently the ArcGIS Maps SDK for .NET does not
```
Comment (pMaske) [inline src/MAUI/Maui.Samples/Samples/Layers/RasterLayerRasterFunction/RasterLayerRasterFunction.xaml.cs:115]: If this general statement across all SDKs then we should say ArcGIS Maps SDK for Native Apps

File: src/MAUI/Maui.Samples/Samples/Search/FindAddress/FindAddress.xaml.cs [hunk 2]
```diff
@@ -139,7 +139,7 @@ private async Task<Graphic> GraphicForPoint(MapPoint point)
             // Get image as a stream from the resources
             // Picture is defined as EmbeddedResource and DoNotCopy
             Stream resourceStream = currentAssembly.GetManifestResourceStream(
-                "ArcGISRuntimeMaui.Resources.PictureMarkerSymbols.pin_star_blue.png");
+                "ArcGISMapsSDK.Resources.PictureMarkerSymbols.pin_star_blue.png");
```
Comment (pMaske) [inline src/MAUI/Maui.Samples/Samples/Search/FindAddress/FindAddress.xaml.cs:142]: We should avoid using `ArcGISMapsSDK` here. Perhaps need to think of a better name.

File: src/MAUI/Maui.Samples/Helpers/ArcGISLoginPrompt.cs [hunk 3]
```diff
@@ -25,13 +25,13 @@
...
 
-namespace ArcGISRuntimeMaui.Helpers
+namespace ArcGIS.Helpers
 {
     internal static class ArcGISLoginPrompt
     {
         private const string ArcGISOnlineUrl = "https://www.arcgis.com/sharing/rest";
 
-        // - The Client ID for an app registered with the server (the ID below is for a public app created by the ArcGIS Runtime team).
+        // - The Client ID for an app registered with the server (the ID below is for a public app created by the ArcGIS Maps SDK team).
```
Comment (pMaske) [inline src/MAUI/Maui.Samples/Helpers/ArcGISLoginPrompt.cs:34]:
```suggestion
        // - The Client ID for an app registered with the server (the ID below is for a public app created by the ArcGIS Maps SDK  for Native Apps team).
```

File: src/MAUI/Maui.Samples/Resources/SettingsPage/about.md [hunk 4]
```diff
@@ -1,6 +1,6 @@
-﻿# ArcGIS Runtime SDK Samples (.NET MAUI)
+# ArcGIS Maps SDK Samples (MAUI)
```
Comment (pMaske) [inline src/MAUI/Maui.Samples/Resources/SettingsPage/about.md:1]:
```suggestion
# ArcGIS Maps SDK Samples for .NET (.NET MAUI)
```

**Review by amweis226 — APPROVED** [review]

**Review by pMaske — CHANGES_REQUESTED** [review]
@ZackAllen thanks for updating this.
I have few comments and suggestions but may have missed to comment on all instances because PR is too big
Please see the comments and apply them to places wherever applicable.

**Pull request merged**